cypress/integration/brewery/ScenarioSharing.js

```diff
@@ -1,40 +1,211 @@
+      ACLSecurity: [
+        { id: USERS_LIST[1].email, role: ROLES.SCENARIO.VIEWER },
+        { id: USERS_LIST[2].email, role: ROLES.SCENARIO.EDITOR },
+        { id: USERS_LIST[3].email, role: ROLES.SCENARIO.VALIDATOR },
+      ],
+    };
+    RolesEdition.confirmNewPermissions(expectedNewPermissions);
+
+    RolesEdition.getShareScenarioButton().click();
+    RolesEdition.getRoleEditorAgentName(USERS_LIST[1].email).should('have.text', USERS_LIST[1].email);
+    RolesEdition.getSelectWithAction(USERS_LIST[1].email).find('input').should('value', ROLES.SCENARIO.VIEWER);
+    RolesEdition.getRoleEditorAgentName(USERS_LIST[2].email).should('have.text', USERS_LIST[2].email);
+    RolesEdition.getSelectWithAction(USERS_LIST[2].email).find('input').should('value', ROLES.SCENARIO.EDITOR);
+    RolesEdition.getRoleEditorAgentName(USERS_LIST[3].email).should('have.text', USERS_LIST[3].email);
+    RolesEdition.getSelectWithAction(USERS_LIST[3].email).find('input').should('value', ROLES.SCENARIO.VALIDATOR);
   });
 
-  it('is shown a message error when unauthorized changes are tried in scenario sharing pop-up', () => {
-    // TODO: open share pop-up and check error messages are displayed when unauthorized changes are made:
-    //  - remove the last admin of a scenario
-    //  - ...
+  it('Check Viewer permissions', () => {
+    Scenarios.getScenarioValidateButton().should('not.exist');
+
+    RolesEdition.getShareScenarioButton().click();
+    RolesEdition.getShareScenarioDialogAgentsSelect().should('not.exist');
+    RolesEdition.getRoleEditorAgentName(USERS_LIST[1].email).should('have.text', USERS_LIST[1].email);
+    RolesEdition.getSelectWithAction(USERS_LIST[1].email).find('input').should('value', ROLES.SCENARIO.VIEWER);
+    RolesEdition.getSelectWithAction(USERS_LIST[1].email).find('div').should('have.attr', 'aria-disabled');
+    RolesEdition.getRoleEditorAgentName(USERS_LIST[2].email).should('have.text', USERS_LIST[2].email);
+    RolesEdition.getSelectWithAction(USERS_LIST[2].email).find('input').should('value', ROLES.SCENARIO.EDITOR);
+    RolesEdition.getSelectWithAction(USERS_LIST[2].email).find('div').should('have.attr', 'aria-disabled');
+    RolesEdition.getRoleEditorAgentName(USERS_LIST[3].email).should('have.text', USERS_LIST[3].email);
+    RolesEdition.getSelectWithAction(USERS_LIST[3].email).find('input').should('value', ROLES.SCENARIO.VALIDATOR);
+    RolesEdition.getSelectWithAction(USERS_LIST[3].email).find('div').should('have.attr', 'aria-disabled');
+    RolesEdition.getSelectWithAction('Workspace').find('div').should('have.attr', 'aria-disabled');
+    RolesEdition.getShareScenarioDialogConfirmAddAccessButton().should('not.exist');
+    RolesEdition.getShareScenarioDialogFirstCancelButton().click();
+
+    Scenarios.getScenarioValidateButton().should('not.exist');
+    Scenarios.getScenarioRejectButton().should('not.exist');
+    Scenarios.getScenarioCreationButton().should('be.visible').should('be.disabled');
```

Comment:
The scenario creation button should not be checked inside this test (idem for other roles), its availability depends on the role of the user in the workspace (we're checking roles at the scenario level here)